readthedocs/templates/projects/environmentvariable_form.html

```diff
@@ -0,0 +1,28 @@
+{% extends "projects/project_edit_base.html" %}
+
+{% load i18n %}
+
+{% block title %}{% trans "Environment Variables" %}{% endblock %}
+
+{% block nav-dashboard %} class="active"{% endblock %}
+
+{% block editing-option-edit-environment-variables %}class="active"{% endblock %}
+
+{% block project-environment-variables-active %}active{% endblock %}
+{% block project_edit_content_header %}{% trans "Environment Variables" %}{% endblock %}
+
+{% block project_edit_content %}
+  <p>
+    {% blocktrans trimmed %}
+      Notice that the values are not escaped when your builds are executed.
+      Special characters (for bash) should be escaped accordingly.
```

Comment:
I think we should escape these sorts of things. Think a long random secret key that might have some special characters like `$` in it. The user should be able to just copy and paste it in.

Reply:
Agreed, let's make it easier for the user. This is a technical implementation detail that the user shouldn't have to worry about

Reply:
Good point.

I don't want to do this by myself. I checked the code and I found that we are already doing something similar at 

https://github.com/rtfd/readthedocs.org/blob/089753253397a0399175969e2e028eecc3b74dd2/readthedocs/doc_builder/environments.py#L294-L295

I found that there is [`shlex.quote`](https://docs.python.org/3/library/shlex.html#shlex.quote) in Python3 and [`pipes.quote`](https://docs.python.org/2/library/pipes.html#pipes.quote) in Python2. I will use this accordingly.

> NOTE: we may want to replace our `bash_escape_re` regex also.

Reply:
I tested this with,

* enter: `{1}\|\1+-(8)$`
* results: `'{1}\\|\\1+-(8)$'`

and,

* enter: `alo'ja`
* results: `\'alo\'"\'"\'ja\'`

are they correct?